log_entry.go

```diff
@@ -0,0 +1,242 @@
+package logrus
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"time"
+)
+
+type formatMode int
+
+const (
+	formatted formatMode = iota
+	unformatted
+	newLine
+)
+
+// An entry is the final or intermediate Logrus logging entry. It contains all
+// the fields passed with WithField{,s}. It's finally logged when Debug, Info,
+// Warn, Error, Fatal or Panic is called on it. These objects can be reused and
+// passed around as much as you wish to avoid field duplication.
+type LogEntry struct {
+	Logger *LogWriter
+
+	// Contains all the fields set by the user.
+	Data Fields
+
+	// Time at which the log entry was created
+	Time time.Time
+
+	// Level the log entry was logged at: Debug, Info, Warn, Error, Fatal or Panic
+	// This field will be set on entry firing and the value will be equal to the one in Logger struct field.
+	Level Level
+
+	// Message passed to Debug, Info, Warn, Error, Fatal or Panic
+	Message string
+
+	// When formatter is called in entry.log(), an Buffer may be set to entry
+	Buffer *bytes.Buffer
```

Comment:
I believe the only reason they put it in the `Entry` type was to pool it. I think we will get more memory allocations when we log the same entry multiple times if we take it out! 

But with the new design, we log a NEW entry object almost every time we call `Write` (because usually every `Write` operation follows a call to an `AsXYZ` method which will new the Entry). I think it makes sense to take it out.